src/system/SystemMutex.h

```diff
@@ -100,10 +98,28 @@ class DLL_EXPORT Mutex
     Mutex & operator=(const Mutex &) = delete;
 };
 
-inline Mutex::Mutex() {}
+class ScopedMutexLock
```

Comment:
It looks ScopedMutexLock is same as lock_guard which is a mutex wrapper that provides a convenient RAII-style mechanism for owning a mutex for the duration of a scoped block. What is the benefits to define our own version?

Reply:
`std::lock_guard` doesn't work with `chip::System::Mutex` as it is currently defined. But perhaps revising our `Mutex` is a better approach.